signature/steps.go

```diff
@@ -45,8 +50,8 @@ func SignSteps(s pipeline.Steps, key jwk.Key, env map[string]string, repoURL str
 }
 
 // SignPipeline adds signatures to each command step (and recursively to any command steps that are within group steps) within a pipeline
-func SignPipeline(p *pipeline.Pipeline, key jwk.Key, repo string) error {
-	if err := SignSteps(p.Steps, key, p.Env.ToMap(), repo); err != nil {
+func SignPipeline(p *pipeline.Pipeline, key jwk.Key, repo string, logger Logger, debugSigning bool) error {
+	if err := SignSteps(p.Steps, key, repo, WithEnv(p.Env.ToMap()), WithLogger(logger), WithDebugSigning(debugSigning)); err != nil {
```

Comment:
This seems a bit weird, we aren't re-using the functional options like in SignSteps. Is it because the environment is passed in via the `p.Env` and not directly as a variable?

Could we just update the [three places Agent calls this method](https://github.com/search?q=repo%3Abuildkite%2Fagent%20SignPipeline&type=code) to instead pass the environment explicitly, since we don't need the full Pipeline?

That way the function signature becomes:
```
func SignPipeline(key jwk.Key, repo string, opts ...Option) error
```

Reply:
I was pairing with Benno at the time and he suggested doing as above. I'll switch it over to using options.

Reply:
The steps still need to be passed in though right?

```go
func SignPipeline(s pipeline.Steps, key jwk.Key, repo string, opts ...Option) error
```